File: src/main/java/me/devsnox/playtime/connection/SyncMySQL.java
package me.devsnox.playtime.connection;

import java.sql.*;

/**
 * Copyright by DevSnox
 * E-Mail: me.devsnox@gmail.com
 * Skype: DevSnox
 */
public class SyncMySQL {

    private Connection connection;
    private SpigotConnection spigotConnection;

    public SyncMySQL(SpigotConnection varoxConnection, Connection connection) {
        this.connection = connection;
        this.spigotConnection = varoxConnection;
    }

    public void update(String statment) {
        refresh();
        if (statment != null) {
            try {
                Statement st = this.connection.createStatement();
                st.executeUpdate(statment);
                st.close();
            } catch (SQLException e) {
                e.printStackTrace();
            }

        }
    }

    public void preparedUpdate(String statment) {
        refresh();
        if (statment != null) {
            try {
                PreparedStatement st = this.connection.prepareStatement(statment);
                st.executeUpdate(statment);
                st.close();
            } catch (SQLException e) {
                e.printStackTrace();
            }

        }
    }

    public void preparedUpdate(PreparedStatement statment) {
        refresh();
        if (statment != null) {
            try {
                statment.executeUpdate();
            } catch (SQLException e) {
                e.printStackTrace();
            }

        }
    }

    public ResultSet query(String statment) {
        refresh();
        if (connection != null) {
            if (statment == null) {
                return null;
            } else {
                ResultSet resultSet = null;

                try {
                    Statement st = this.connection.createStatement();
                    resultSet = st.executeQuery(statment);
                } catch (SQLException e) {
                    e.printStackTrace();
                }

                return resultSet;
            }
        }
        return null;
    }

    private void refresh() {
        try {
            if (!this.connection.isValid(2) || this.connection.getMetaData() == null) {
                this.spigotConnection.disconnect();
                this.spigotConnection.connect();
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/java/me/devsnox/playtime/connection/SyncMySQL.kt
package me.devsnox.playtime.connection

import java.sql.Connection
import java.sql.PreparedStatement
import java.sql.ResultSet
import java.sql.SQLException

/**
 * Created by Yasin Dalal
 * Copyright (c) 2017-2019 Yasin Dalal
 * GitHub: https://github.com/DevSnox
 * E-Mail: yasin@dalal.ch
 */
@Deprecated("")
class SyncMySQL(private val spigotConnection: SpigotConnection, private val connection: Connection?) {
    fun update(statment: String?) {
        refresh()
        if (statment != null) {
            try {
                val st = connection!!.createStatement()
                st.executeUpdate(statment)
                st.close()
            } catch (e: SQLException) {
                e.printStackTrace()
            }
        }
    }

    fun preparedUpdate(statment: String?) {
        refresh()
        if (statment != null) {
            try {
                val st = connection!!.prepareStatement(statment)
                st.executeUpdate(statment)
                st.close()
            } catch (e: SQLException) {
                e.printStackTrace()
            }
        }
    }

    fun preparedUpdate(statment: PreparedStatement?) {
        refresh()
        if (statment != null) {
            try {
                statment.executeUpdate()
            } catch (e: SQLException) {
                e.printStackTrace()
            }
        }
    }

    fun query(statment: String?): ResultSet? {
        refresh()
        return if (connection != null) {
            if (statment == null) {
                null
            } else {
                var resultSet: ResultSet? = null
                try {
                    val st = connection.createStatement()
                    resultSet = st.executeQuery(statment)
                } catch (e: SQLException) {
                    e.printStackTrace()
                }
                resultSet
            }
        } else null
    }

    private fun refresh() {
        try {
            if (!connection!!.isValid(2) || connection.metaData == null) {
                spigotConnection.disconnect()
                spigotConnection.connect()
            }
        } catch (e: SQLException) {
            e.printStackTrace()
        }
    }

}